diff -rup system-config-language-1.3.4/src/language_backend.py system-config-language-1.3.4_mod/src/language_backend.py
--- system-config-language-1.3.4/src/language_backend.py	2010-03-09 17:03:43.000000000 +0530
+++ system-config-language-1.3.4_mod/src/language_backend.py	2010-08-11 09:46:23.000000000 +0530
@@ -66,12 +66,12 @@ class LanguageBackend:
             return defaultLang, langs
 
     def removeEncoding(self, lang):
-        if '.' in lang:
-            langBase = string.split(lang, '.')
-            return langBase[0]
-        elif '@' in lang:
+        if '@' in lang:
             langBase = string.split(lang, '@')
             return langBase[0]
+        elif '.' in lang:
+            langBase = string.split(lang, '.')
+            return langBase[0]
         else:
             return lang
 
